File: SeatGeekSearch/app/src/main/java/com/cjwilliams24680/seatgeeksearch/models/EventSearchMetadata.java
package com.cjwilliams24680.seatgeeksearch.models;

import android.support.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Created by chris on 4/11/18.
 */

public class EventSearchMetadata {

    @Expose
    @SerializedName("per_page")
    @NonNull
    Integer perPage;

    @Expose
    @SerializedName("total")
    @NonNull
    Integer total;

    @Expose
    @SerializedName("page")
    @NonNull
    Integer page;

    public EventSearchMetadata() { }

    @NonNull
    public Integer getPerPage() {
        return perPage;
    }

    @NonNull
    public Integer getTotal() {
        return total;
    }

    @NonNull
    public Integer getPage() {
        return page;
    }
}

File: SeatGeekSearch/app/src/main/java/com/cjwilliams24680/seatgeeksearch/network/models/EventSearchMetadata.kt
package com.cjwilliams24680.seatgeeksearch.network.models

import com.google.gson.annotations.SerializedName

/**
 * Created by chris on 4/11/18.
 */
data class EventSearchMetadata(
    @SerializedName("per_page")
    val perPage: Int,
    @SerializedName("total")
    val total: Int,
    @SerializedName("page")
    val page: Int
)
